### PR TITLE
fix: Example pre-commit hook adding cached examples

### PR DESCRIPTION
**Pull request by akocbek**
### Description

Do not add examples from .tfsec folder

**Check the relevant boxes:**
- [x] Bug fix (nonbreaking change that fixes an issue)
- [ ] New feature (nonbreaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [ ] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

File: module-assets/ci/terraformDocExamples.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def get_readme_titles():
     readme_titles = []
     for readme_file in Path("examples").rglob("README.md"):
         path = str(readme_file)
-        if not (".terraform") in path:
+        if not (".terraform") in path and not (".tfsec") in path:
```
Comment (ocofaigh) [inline module-assets/ci/terraformDocExamples.py:19]: should we just ignore all hidden folders? So any folder that starts with a dot ?

Comment (akocbek) [reply module-assets/ci/terraformDocExamples.py]: good catch. updated

File: module-assets/ci/terraformDocExamples.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ def get_readme_titles():
     readme_titles = []
     for readme_file in Path("examples").rglob("README.md"):
         path = str(readme_file)
-        if not (".terraform") in path:
+        if not ("/.") in path:
```
Comment (ocofaigh) [inline module-assets/ci/terraformDocExamples.py:19]: hmm, would this catch `.terraform` directory though?

Comment (akocbek) [reply module-assets/ci/terraformDocExamples.py:19]: yes, catches both `.terraform` and `.tfsec`. we scan only files inside `examples` folder

Comment (daniel-butler-irl) [reply module-assets/ci/terraformDocExamples.py:19]: Here is to check if a path contains a hidden directory, its just a quick mockup so modify as needed
```python
def in_hidden_dir(filepath, hidden=False):
    if not hidden:
        parent = str(filepath.parent.absolute())
        parent_name = filepath.parent.absolute().name
        if parent_name != "":
            if parent_name.startswith("."):
                return True
            else:
                return in_hidden_dir(Path(parent), hidden)
        else:
            return False
    else:
        return True
```

Comment (ocofaigh) [reply module-assets/ci/terraformDocExamples.py:19]: agreed on scrum current approach is ok to use

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**